students/Hasan/Lab2/Part1_HomebrewPWM/PWMModified/PWMModified/PWMModified.ino

```diff
@@ -0,0 +1,54 @@
+const int gLED = 10; 
+
+
+void setup() {
+  // put your setup code here, to run once:
+   pinMode(gLED, OUTPUT); 
+}
+
+void pwm(int pin, double duty_cycle, double duration) {
+
+// In seconds
+  double Current_time = millis()/1000; 
+
+  // Check if the time is less than the duration 
+  if (Current_time < duration){
+      //Calculate Ton
+      
+  double Ton; 
+  //Calculate Toff
+  double Toff; 
+
+// Remember microseconds must be divided by 1000
+  Ton = duty_cycle * 10/1000.0; 
+
+  Toff = abs(Ton - (10/1000.0)); 
+
+
+ //boolean logic
+ // Need to multiply seconds by 1000000
+  digitalWrite(pin, duty_cycle && HIGH); 
+
+  delayMicroseconds(Ton*1000000); 
+
+  digitalWrite(pin, duty_cycle && LOW); 
```

Comment:
I'm not sure this logic is doing what you want it to. It works for the `HIGH` case, but here `LOW` is always 0, so the second argument is always 0.